build.sbt

```diff
@@ -460,7 +460,7 @@ lazy val clientTestkit = libraryCrossProject("client-testkit")
       nettyCodecHttp,
     )
   )
-  .dependsOn(client, theDsl)
+  .dependsOn(client, theDsl, server, tests % Test)
```

Comment:
Raising this one for other maintainers: I proposed that we add this to the `client-testkit` module, and add a `server` dependency to that. I wasn't sure where else we can put this utility that has both a server and client dep.

Stan suggested that we move `WebSocketBuilder` to `core` , which we can consider for 1.0. Interestingly, Chris has also suggested that `Client` would be useful in `core` as well. Not sure if that extends to `WSClient`. Thoughts?

Also the plan for 1.0 is to reunify the different WS frame models. Lots of cleanup to do :)

Reply:
`Client` or the whole client package?  Moving just `Client` seems odd, but I haven't heard the rationale.

Reply:
One argument would be that `Client` is just as "primitive" as `HttpApp` or `HttpRoutes`. Also so we can avoid this silliness :)

https://github.com/http4s/http4s/blob/a5397890192c2088c4551058bb14c8be6897bf04/ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Client.scala#L333-L334

---

> or the whole client package?

This deserves its own discussion, but recently I've been daydreaming about collapsing core+server+client+ember into a single http4s artifact. For example would be _really_ nice to get an awesome default `Client` implementation without having to import and muck about with a separate builder.

Reply:
Why is `H2Client` in ember-core instead of ember-client?  Regardless, `TinyClient` is tinier than `Client`, so what would it help?

It's kind of true that `HttpApp` and `HttpRoutes` are the same level as `Client`.  Once server responses become resourceful, they're all views of that "TinyClient" type.  It's just that the server ones happen to be a well known monad transformer and Client is an ad hoc creation of our own.  The argument becomes stronger if the server ones ceased to be an alias to a standard type.

I'll respond to the flattening on the other discussion.